Tests/SwiftSyntaxBuilderTest/DoStmtTests.swift

```diff
@@ -0,0 +1,48 @@
+import XCTest
+import SwiftSyntaxBuilder
+
+final class DoStmtTests: XCTestCase {
+  func testDoStmt() {
+    let buildable = DoStmt(
+      labelName: nil,
+      body: CodeBlock(statementsBuilder: {
+        CodeBlockItem(item: TryExpr(expression: FunctionCallExpr(MemberAccessExpr(base: "a", name: "b"))))
+      }),
+      catchClauses: [
+        CatchClause(CatchItems {
+          CatchItem(pattern: "Error1")
+          CatchItem(pattern: "Error2")
+        }) {
```

Comment:
I’m not entirely convinced this is much better than the default implementation that’s already provided.

```suggestion
        CatchClause(catchItems: CatchItemList([
          CatchItem(pattern: "Error1"),
          CatchItem(pattern: "Error2")
        ])) {
```

Do you have some example where the result builder syntax is superior than using an array here?

Reply:
1. Result builder automatically inserts trailing commas
1. This `CatchClause` convenience initializer adds trailing space to the `catch` keyword

Reply:
1. I really don’t like the idea of introducing yet another type for catch items. We now have `CatchItemsListSyntax`, `CatchItemsList` and `CatchItems`. What do you think about adding an initializer like the following to `CatchItemList`. We could add the same for all syntax collections:
```swift
public init(@CatchItemListBuilder itemsBuilder: () -> ExpressibleAsCatchItemList) {
  self = itemsBuilder().createCatchItemList()
}
```
2. Can’t we just mark the `catch` keyword as `requires_trailing_space` in https://github.com/apple/swift/blob/5dfa894ea773c1cb82106d831cd4d2df4a837b75/utils/gyb_syntax_support/Token.py#L188?

Reply:
1. Great idea, I've raised #519 for it
2. I actually did the opposite to avoid double space [when there is no catch items](https://github.com/apple/swift-syntax/pull/511/files#diff-36fc9e19ca4760fd7343a8a7b18a4b2759cc0a821e1396ec5f804cde41476de0R43), so I'd like to keep this convenience initializer